File: go/raindrops/raindrops.go
package raindrops

import "fmt"

// Convert takes an integer and returns a string.
// If the number has factors of 3, 5, 7, build and return an approptiate string.
// If no factors, pass through the origninal number as a string.
func Convert(num int) string {
	res := ""
	if num%3 == 0 {
		res += "Pling"
	}
	if num%5 == 0 {
		res += "Plang"
	}
	if num%7 == 0 {
		res += "Plong"
	}
	if res == "" {
		return fmt.Sprintf("%d", num)
	}
	return res
}

File: csharp/raindrops/Raindrops.cs
using System;
using System.Collections.Generic;
using System.Text;

public static class Raindrops
{
    private static Dictionary<int, string> map = new Dictionary<int, string>(){
        { 3, "Pling" },
        { 5, "Plang" },
        { 7, "Plong" },
    };
    public static string Convert(int number)
    {
        StringBuilder s = new StringBuilder();
        foreach(KeyValuePair<int, string> entry in map) {
            if (number % entry.Key == 0) {
                s.Append(entry.Value);
            }
        }
        return s.Length == 0 ? number.ToString() : s.ToString();
    }
}